File: src/main/java/com/maxciv/infer/plugin/tree/CellRenderer.java
package com.maxciv.infer.plugin.tree;

import com.intellij.openapi.util.IconLoader;
import com.intellij.ui.ColoredTreeCellRenderer;
import org.jetbrains.annotations.NotNull;

import javax.swing.*;

/**
 * @author maxim.oleynik
 * @since 01.12.2018
 */
public class CellRenderer extends ColoredTreeCellRenderer {

    static final Icon ICON = IconLoader.getIcon("/icons/testPassed.png");

    public CellRenderer() {
        super();
    }

    @Override
    public void customizeCellRenderer(
            @NotNull JTree tree,
            Object value,
            boolean selected,
            boolean expanded,
            boolean leaf,
            int row,
            boolean hasFocus
    ) {
        if (value instanceof TreeNodeData) {
            ((TreeNodeData) value).render(this);
        }
    }
}

File: src/main/kotlin/com/maxciv/infer/plugin/ui/tree/CellRenderer.kt
package com.maxciv.infer.plugin.ui.tree

import com.intellij.ui.ColoredTreeCellRenderer
import javax.swing.JTree

/**
 * @author maxim.oleynik
 * @since 01.12.2018
 */
class CellRenderer : ColoredTreeCellRenderer() {

    override fun customizeCellRenderer(
        tree: JTree,
        value: Any,
        selected: Boolean,
        expanded: Boolean,
        leaf: Boolean,
        row: Int,
        hasFocus: Boolean
    ) {
        if (value is TreeNodeData) {
            value.render(this)
        }
    }
}
